File: code/cpp/ch18/practice/warmup_02_jump_game.cpp
/*
 * Warmup 2: Jump Game I
 * Chapter 18: Greedy Algorithms — The Smart Shortcut
 *
 * PROBLEM: Can you reach the last index?
 * EXAMPLES: solve({2,3,1,1,4}) -> true
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */

#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

bool solve(vector<int> nums) {
    // TODO: Replace this with your solution
    return false;
}

// -- Do not change anything below this line --------------------------
int main() {
    int n; cin >> n;
    vector<int> nums(n);
    for (int i = 0; i < n; i++) cin >> nums[i];
    cout << (solve(nums) ? "true" : "false") << endl;
    return 0;
}

File: code/python/ch18/practice/warmup_02_jump_game.py
"""
Warmup 2: Jump Game I
======================
Chapter 18: Greedy Algorithms — The Smart Shortcut

PROBLEM
-------
Given an array of non-negative integers, you start at index 0.
Each element is your maximum jump length from that position.
Can you reach the last index?

EXAMPLES
--------
>>> solve([2, 3, 1, 1, 4])
True
>>> solve([3, 2, 1, 0, 4])
False

CONSTRAINTS
-----------
- 1 <= len(nums) <= 10^4
- 0 <= nums[i] <= 10^5
"""


def solve(nums: list[int]) -> bool:
    """Return True if you can reach the last index."""
    pass  # TODO: Replace this with your solution


# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    nums = list(map(int, input().strip().split()))
    print(solve(nums))
